### ISSUE TITLE
First layer with single channel

### ISSUE THREAD

**Issue by sinuku** [pytorch/vision#4297]
## 🚀 Feature

When calling the model, it would be nice to be able to set the number of channels in the first layer.

## Motivation

When using a grayscale image, I want the input to be a single channel.

## Pitch

ex)
```
self.conv1 = nn.Conv2d(3, self.inplanes, kernel_size=7, stride=2, padding=3, bias=False)
-> self.conv1 = nn.Conv2d(1, self.inplanes, kernel_size=7, stride=2, padding=3, bias=False)

layers.append(ConvBNActivation(3, firstconv_output_channels, kernel_size=3, stride=2, norm_layer=norm_layer, activation_layer=nn.Hardswish))
-> layers.append(ConvBNActivation(1, firstconv_output_channels, kernel_size=3, stride=2, norm_layer=norm_layer,  activation_layer=nn.Hardswish))
```

## Alternatives

Use parameters when calling the model.
ex)
`self.conv1 = nn.Conv2d(input, self.inplanes, kernel_size=7, stride=2, padding=3, bias=False)`

**Comment by datumbox** [comment]
I think the trade off here is adding one more parameter in all model constructor methods VS keeping the API simple. The reason we don't support the extra parameter is because it is straight forward to achieve what you want with a bit of model surgery (like on the examples you provide). 

As a side-note, simply changing the first convolution will work when the number of channels is close to 3. If the number is significantly larger you might start facing issues as you might need to adjust the width parameter of the network to account for the extra information of the input.

**Comment by someshfengde** [comment]
yes it'll be good for adding the parameter as the number of channels but the thing is it'll not keep API simple 

completely agree with @datumbox
